src/zmalloc.c

```diff
@@ -388,7 +415,11 @@ char *zstrdup(const char *s) {
 }
 
 size_t zmalloc_used_memory(void) {
-    size_t um = atomic_load_explicit(&used_memory, memory_order_relaxed);
+    assert(total_active_threads < MAX_THREADS_NUM);
+    size_t um = 0;
+    for (int i = 0; i < total_active_threads; i++) {
+        um += used_memory_thread[i];
+    }
```

Comment:
I am not sure I get this part. without atomic operations on both readers and writers, I think we will get stale data?

Reply:
@PingXie Yes, the worst case is when read thread is reading while the write thread are writing to thread-local storage variable, but the data will not lose, just stale.

Reply:
That makes sense. so we are essentially saying/assuming the staleness is (tightly) bounded, which I think is reasonable.